imports/api/accounts/methods.js

```diff
@@ -131,6 +131,45 @@ export const insertAccountOnSignUp = new ValidatedMethod({
 });
 
 
+export const updateProfile = new ValidatedMethod({
+    name: 'updateProfile',
+    mixins: [LoggedInMixin],
+    checkLoggedInError: {
+        error: 'notLogged',
+        message: 'You need to be logged in to update profile'
+    },
+    validate: new SimpleSchema({
+        'users.name': {
+            type: String
+        },
+        'users.username': {
+            type: String
+        },
+        'users.number': {
+            type: String
+        },
+        'users.email': {
+            type: String
+        },
+        'users.address': {
+            type: String
+        }
+    }).validator(),
+    run({ users }) {
+        Meteor.users.update(
```

Comment:
Add a condition which will check that if username/email is empty, should throw error that you can't empty the username/email.